### DISCUSSION TITLE
Update the Mautic Twitter integration

### DISCUSSION THREAD

**Discussion by RCheesley** [mautic/mautic#10429] [Requests For Proposals (RFPs)]
## 🤔  Problem outline
Since Twitter updated their API authentication settings, the Twitter integration is still relying on the very outdated OAuth1.

This RFP is a project to update the Twitter integration to take into account the new authentication configurations (and maybe to take into account updates coming soon to the API).

Note: there is a new API coming in soon: https://developer.twitter.com/en/products/twitter-api/early-access/guide 

## ✅  Key requirements

- Update the current integration to use OAuth2
- Update the end user documentation to ensure it is relevant and includes updated screenshots
- Write an article for the knowledgebase on how to configure and use the integration
- Update the developer documentation if required

## 💰 Funding

[![open collective badge](https://opencollective.com/rfp-general-fund/tiers/fix-twitter-integration/badge.svg?label=Fix%20Twitter%20Integration&color=brightgreen)](https://opencollective.com/rfp-general-fund/contribute/fix-twitter-integration-32209/checkout)

This project is not currently funded. If you would like to contribute toward this project, please make a donation to the [RFP General Fund](https://opencollective.com/rfp-general-fund) project on Open Collective and select the '[Fix Twitter Integration](https://opencollective.com/rfp-general-fund/contribute/fix-twitter-integration-32209/checkout)' tier.

## ℹ️  Timelines

This should be targeted for a minor release, as it will be an enhancement of an existing feature.

## 👉  How to make a proposal

Please review the information in the [Community Handbook](https://contribute.mautic.org/product-team/request-for-proposals-rfp-process#how-to-claim-a-project) regarding the workflow for proposing and working on this project.

**Comment by cloudpotions** [comment]
Hello please refer to https://github.com/mautic/mautic/issues/9042 - There is not a problem with the Twitter plugin, it is actually authentication settings on the Twitter Developer App. I am running Mautic V 4.1.1 and it is working perfectly for me. I also made a request today to update documentation on Twitter Mautic Docs... 

For easy reference please also see below:

A Twitter application has to be created for authorization. To create/manage one, go to https://developer.twitter.com/. Once you have created a Twitter App, you can edit the user Authentication settings. It is important to enable OAuth 1.0a and disable OAuth 2.0 for this plugin to work properly. Furthermore, when enabling OAuth 1.0a on your Twitter App, you will have to enter the Callback URI / Redirect URL. The Twitter Plugin in Mautic will show you this URL so you can copy and paste it into your Twitter App.
